### PR TITLE
Fix loss function bug in GradientTape example

### PR DESCRIPTION
**Pull request by fibigerg**
Add from_logits=True to loss function in tf.GradientTape example. The code is broken without it.

ALLOW_H5=true

### CODE REVIEW

**Comment by fibigerg** [comment]
I dig deeper into this, comparing model.compile/fit and GradientTape learning progress on a simple model. There really should be `training=True` in the model call. I am not sure about `from_logits=True`. I had to remove it to have exactly the same learning curve as with model.compile/fit approach.

File: site/en/guide/keras/train_and_evaluate.ipynb [hunk 1]
```diff
@@ -1575,7 +1575,7 @@
         "# Instantiate an optimizer.\n",
         "optimizer = keras.optimizers.SGD(learning_rate=1e-3)\n",
         "# Instantiate a loss function.\n",
-        "loss_fn = keras.losses.SparseCategoricalCrossentropy()\n",
+        "loss_fn = keras.losses.SparseCategoricalCrossentropy(from_logits=True)\n",
```
Comment (yashk2810) [inline site/en/guide/keras/train_and_evaluate.ipynb]: The last layer activation of the model is set to 'softmax'. Can you remove that so that from_logits=True will work?
